### PR TITLE
Create Kytos API Authentication blueprint

### PR DESCRIPTION
**Pull request by gleybersonandrade**

### CODE REVIEW

File: docs/blueprints/EP018.rst [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+Gleyberson Andrade and Carlos Magno
+
+Approvers (PTL)
+---------------
+
+Version
+-------
+0.0.1
+
+
```
Comment (beraldoleal) [inline docs/blueprints/EP018.rst]: @gleybersonandrade @cmagnobarbosa Please, can you use the new header model, proposed by @hdiogenes ?

File: docs/blueprints/EP018.rst [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+The endpoint restriction request happens when a decorator is inserted, ensuring
+access to only authenticated users. This endpoint can be accessed through a
+token, generated by the module after the user authentication request. It is
+not the purpose of this module to take care of the authorization part
+(access level).
+
+Motivation
+==========
+
+Today, all NApps rest endpoints that are created through decorator @rest are
```
Comment (beraldoleal) [inline docs/blueprints/EP018.rst]: REST in upper case

File: docs/blueprints/EP018.rst [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+publicly exposed, so unauthorized persons or applications who have access to
+these endpoints can change Napps behavior. If want to restrict this access,
```
Comment (beraldoleal) [inline docs/blueprints/EP018.rst]: If the developer want to...

File: docs/blueprints/EP018.rst [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+mechanisms from the NApp developer.
+
+This proposal provides an authentication mechanism to the NApp developer, so
+each developer doesn't have to create their own mechanism.
+
+Specification
+=============
+
+From the NApps developer point of view, he continues to use decorator
+@rest('endpoint'). In this project, we propose the addition of a new decorator,
```
Comment (beraldoleal) [inline docs/blueprints/EP018.rst]: In this blueprint

File: docs/blueprints/EP018.rst [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+which determines whether the endpoint will be public or private. By default,
+the endpoint will be public. If the user wants to create a private endpoint,
+he must use the @authenticated decorator as shown:
```
Comment (beraldoleal) [inline docs/blueprints/EP018.rst]: he/she or a neutral pronoun.

File: docs/blueprints/EP018.rst [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+- GET /api/kytos/auth/v1/users/ - This endpoint lists the registered users.
+
+::
+
+    $ curl -X GET \
+        -H 'Accept:application/json' \
+        -H "Authorization: Bearer ${TOKEN}"\
+        URL
+
+- GET /api/kytos/auth/v1/users/<user_id>/ - This endpoint lists a specific user.
```
Comment (beraldoleal) [inline docs/blueprints/EP018.rst]: Get details about a specific user.

**Review by beraldoleal — CHANGES_REQUESTED** [review]

File: docs/blueprints/EP018.rst [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+Endpoints REST:
+---------------
+
+- POST /api/kytos/auth/v1/users/ - This endpoint creates new users.
+
+::
+
+    $ curl -X POST \
+        -H "Content-Type: application/json" \
+        -d '{"username":"babel42", "password":"youshallnotpass", “email”: “babel42@email.com”}'
```
Comment (hdiogenes) [inline docs/blueprints/EP018.rst]: Fix quotes: s/“email”/"email"/, etc.

File: docs/blueprints/EP018.rst [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+@rest('endpoint'). In this blueprint, we propose the addition of a new
+decorator, which determines whether the endpoint will be public or private.
+By default, the endpoint will be public. If users want to create a private
+endpoint, they must use the @authenticated decorator as shown:
+
+.. code:: python
+
+    from kytos.core.decorators import rest, authenticated
+    
+    @rest(‘endpoint’)
```
Comment (hdiogenes) [inline docs/blueprints/EP018.rst]: Fix quotes here

**Review by beraldoleal — APPROVED** [review]

File: docs/blueprints/EP018.rst [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+        - GET /api/kytos/auth/v1/users/ - List all users.
+        - GET /api/kytos/auth/v1/users/<user_id>/ - List specific user.
+        - DELETE /api/kytos/auth/v1/users/<user_id>/ - Delete specific user.
+        - PATCH /api/kytos/auth/v1/users/<user_id>/ - Update specific user.
+
+Open Issues
+===========
+
+- `Start Blueprint: Kytos API Authentication <https://github.com/kytos/kytos/issues/861>`_
+- `Create Kytos API Authentication blueprint <https://github.com/kytos/kytos/pull/955>`_
```
Comment (hdiogenes) [inline docs/blueprints/EP018.rst]: Move these links to the References, and add here any questions that are still being discussed, if any.

**Pull request merged**